### PR TITLE
refactor: simplify test keeper functions

### PR DESCRIPTION
**Pull request by aljo242**
Closes ##498

### What does this PR does?

Changes the `testutil/keeper` function `AllKeepers()` to: 
```golang
func NewTestSetup(t testing.TB) (sdk.Context, TestKeepers, TestMsgServers)
```

The `TestKeepers` struct just contains all of the keepers:
```golang
type TestKeepers struct {
	CampaignKeeper           *campaignkeeper.Keeper
	LaunchKeeper             *launchkeeper.Keeper
	ProfileKeeper            *profilekeeper.Keeper
	RewardKeeper             *rewardkeeper.Keeper
	MonitoringConsumerKeeper *monitoringcmodulekeeper.Keeper
	BankKeeper               bankkeeper.Keeper
	IBCKeeper                *ibckeeper.Keeper
}
```

The `TestMsgServers` struct just contains all of the message servers:
```golang
type TestMsgServers struct {
	ProfileSrv     profiletypes.MsgServer
	LaunchSrv      launchtypes.MsgServer
	CampaignSrv    campaigntypes.MsgServer
	RewardSrv      rewardtypes.MsgServer
	MonitoringcSrv monitoringcmoduletypes.MsgServer
}
```

The remaining changes apply these changes throughout our tests.

### CODE REVIEW

File: x/campaign/keeper/invariants_test.go
Comment (lubtd) [inline x/campaign/keeper/invariants_test.go:14]: In each module, we have a `setupMsgServer` that returns the message server for the necessary keepers like:
```
keeper.NewMsgServerImpl(*tk.CampaignKeeper),
profilekeeper.NewMsgServerImpl(*tk.ProfileKeeper)
```
Maybe for further simplification we can also add these msgServers in `TestKeepers` as well:
```
type TestKeepers struct {
	CampaignKeeper           *campaignkeeper.Keeper
	LaunchKeeper             *launchkeeper.Keeper
	ProfileKeeper            *profilekeeper.Keeper
	RewardKeeper             *rewardkeeper.Keeper
	MonitoringConsumerKeeper *monitoringcmodulekeeper.Keeper
	BankKeeper               bankkeeper.Keeper
	IBCKeeper                *ibckeeper.Keeper
        LaunchMsgServer    launchkeeper.MsgServer
        ...
}
```

**Review by lubtd — COMMENTED** [review]
We originally had a test keeper constructor method for each module
```
func Campaign(t testing.TB) (*campaignkeeper.Keeper, sdk.Context) {
```
to have some method that allows to instantiate a keeper object without doing the full initialization. However, because of the dependencies between module, we almost always do most of the initialization for all modules.
For simplicity I think we can remove all these methods and use `TestKeepers` everywhere.
The change might be consequent since all unit tests should be changed. We can create an issue for this and handle it once this PR is merged. The priority of the PR is to unblock `participation` initialization

**Comment by aljo242** [comment]
I'll push to get all of these recommended changes into this PR today !  I think it makes sense to just do them all now

**Comment by aljo242** [comment]
The only function I have left is `MonitoringcWithIBCMocks()` which takes additional arguments for IBC mocking.

File: testutil/keeper/keeper.go
Comment (Pantani) [inline testutil/keeper/keeper.go:8]:
```suggestion
	sdk "github.com/cosmos/cosmos-sdk/types"
	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
	ibcclienttypes "github.com/cosmos/ibc-go/v2/modules/core/02-client/types"
	ibcconnectiontypes "github.com/cosmos/ibc-go/v2/modules/core/03-connection/types"
	ibckeeper "github.com/cosmos/ibc-go/v2/modules/core/keeper"
	"github.com/stretchr/testify/require"
	"github.com/tendermint/tendermint/libs/log"
	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"

	campaignkeeper "github.com/tendermint/spn/x/campaign/keeper"
	campaigntypes "github.com/tendermint/spn/x/campaign/types"
	launchkeeper "github.com/tendermint/spn/x/launch/keeper"
	launchtypes "github.com/tendermint/spn/x/launch/types"
	monitoringcmodulekeeper "github.com/tendermint/spn/x/monitoringc/keeper"
	monitoringcmoduletypes "github.com/tendermint/spn/x/monitoringc/types"
	profilekeeper "github.com/tendermint/spn/x/profile/keeper"
	profiletypes "github.com/tendermint/spn/x/profile/types"
	rewardkeeper "github.com/tendermint/spn/x/reward/keeper"
	rewardtypes "github.com/tendermint/spn/x/reward/types"
```

File: testutil/keeper/keeper.go
Comment (Pantani) [inline testutil/keeper/keeper.go:41]:
```suggestion
	MonitoringConsumerKeeper *monitoringckeeper.Keeper
```

File: testutil/keeper/keeper.go
Comment (Pantani) [inline testutil/keeper/keeper.go:52]:
```suggestion
	MonitoringcSrv monitoringctypes.MsgServer
```

File: x/campaign/keeper/grpc_query_mainnet_account_test.go
Comment (Pantani) [inline x/campaign/keeper/grpc_query_mainnet_account_test.go:7]:
```suggestion
	sdk "github.com/cosmos/cosmos-sdk/types"
	"github.com/cosmos/cosmos-sdk/types/query"
	"github.com/stretchr/testify/require"
	"google.golang.org/grpc/codes"
	"google.golang.org/grpc/status"

	testkeeper "github.com/tendermint/spn/testutil/keeper"
	"github.com/tendermint/spn/testutil/sample"
	"github.com/tendermint/spn/x/campaign/keeper"
	"github.com/tendermint/spn/x/campaign/types"
```

File: x/campaign/keeper/invariants_test.go
Comment (Pantani) [inline x/campaign/keeper/invariants_test.go:6]:
```suggestion
	"github.com/stretchr/testify/require"
	
	"github.com/tendermint/spn/testutil/sample"
	testkeeper "github.com/tendermint/spn/testutil/keeper"
	"github.com/tendermint/spn/x/campaign/keeper"
	"github.com/tendermint/spn/x/campaign/types"
```

File: x/campaign/keeper/msg_server_burn_vouchers_test.go
Comment (Pantani) [inline x/campaign/keeper/msg_server_burn_vouchers_test.go:6]:
```suggestion
	sdk "github.com/cosmos/cosmos-sdk/types"
	"github.com/stretchr/testify/require"

	spnerrors "github.com/tendermint/spn/pkg/errors"
	testkeeper "github.com/tendermint/spn/testutil/keeper"
	"github.com/tendermint/spn/testutil/sample"
	"github.com/tendermint/spn/x/campaign/types"
```

File: x/campaign/keeper/msg_server_update_total_supply_test.go
Comment (Pantani) [inline x/campaign/keeper/msg_server_update_total_supply_test.go:6]: If we add the import latter, the `goimport` does not solve the import sort well. It will put the project imports into a new block line. We should fix it manually

**Comment by lubtd** [comment]
> The only function I have left is MonitoringcWithIBCMocks() which takes additional arguments for IBC mocking.

This is fine, this one is very niche

**Review by lubtd — APPROVED** [review]
Awesome👌
Can be merged after Danilo comments and conflicts

**Review by lubtd — APPROVED** [review]

**Pull request merged**
